texastribune/donations-app
# Better deployment + minor updates

#### What's this PR do?
- Uses `aws-cli` to deploy to S3 instead of the weird Ruby gem
- Removes IE 7/8 hacks + build process
- Removes bad tests
- Makes the widget a bit better by clearing the manual input

#### Why are we doing this? How does it help us?
- Deployment is just generally easier and is consistent with that of `code-grabber`
- Browser usage stats in GA don't back up having these cluttering the codebase
- Self-explanatory
- Just usability

#### How should this be manually tested?
- Make sure the following is in your `~/.aws/config`:
```
[default]
aws_access_key_id=YOUR_UNIQUE_ID
aws_secret_access_key=YOUR_SECRET_ACCESS_KEY
```
- `npm install`
- `npm run dev`
- Click any frequency. On the amounts slide, enter an amount in the manual field. Then, click one of the pre-selected amounts. Confirm the value gets cleared out in the manual field.
- Exit the dev server
- `npm run build`
- In `utils/deploy.sh`, add the `--dry-run` flag into each shell command.
- `npm run deploy`
- Confirm there were no errors

#### Have automated tests been added?
No.

#### Are there performance implications? If adding JS can it be done async? Do images/CSS/JS have cache headers set?
NA.

#### What are the relevant tickets?
https://texastribune.atlassian.net/browse/EN-56

#### How should this change be communicated to end users?
NA.

#### Next steps?
NA.

#### Smells?
NA.

#### TODOs:
NA.

#### Has the relevant documentation/wiki been updated?
Yes.

#### Technical debt note
Less.
